### PR TITLE
Knn clustering

### PR DESCRIPTION
**Pull request by karthmnz**
Logic:

1. Find the euclidean distance between from every destination (accident) to every source (ER)
2. Rank the sources to each destination by the distance. 
3. Select the min distance source as a possible candidate. If the capacity is not zero, decrement by 1 else move to the next possible candidate until you are able to decrement by one.

Edge case: When capacity is all zero, (when ER < dest)

In this case, the ER's are already overwhelmed. So, pick the closest (by distance) ER to the incident and assign them to the accident. Flagged as -1. 

The loop logic is similiar to the above.

### CODE REVIEW

File: a_star_algo.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+tags = {
```
Comment (singh264) [inline a_star_algo.py]: - [ ] @karthmnz Add/Use 10 emergency dispatch points as hard-coded list for easy testing

File: a_star_algo.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+df = pd.DataFrame(emergency_locations1) 
+dic = df['osmid'].to_dict()
+nod_ = [dic[a] for a, b in dic.items()] # list of osmid of amenities 
+
+dest = {}
+cap = [2 for i in range(len(nod_))] # cap = capacity at each amenity (hospital + fire-station)
+for i in range(len(nod_)):
+    dest[nod_[i]] = cap[i] # dest = map amenity (hospital/fire-station) osmid to its capacity
+
+# source = [56922195, 29658912, 6894245117, 29656792, 1287736695, 73461503, 267197091, 393521690, 1758011800, 432736587]
```
Comment (karthmnz) [inline a_star_algo.py:20]: Added source @singh264

**Comment by singh264** [comment]
@karthmnz there is an indent error in final block of main.ipynb

```
if (idx != -1):
            if (amenities[i].amenity == 'fire_station'):
            node_colors[idx] = 'white'
            node_size[idx] = 10000
```

**Comment by singh264** [comment]
@karthmnz Requesting demo1/demo2 plot code as shown yesterday in chat

**Comment by karthmnz** [comment]
> 
> 
> @karthmnz Requesting demo1/demo2 plot code as shown yesterday in chat

@singh264 both have been already updated in the main... including the indentation fix

**Pull request closed without merging**